src/mds/Server.cc

```diff
@@ -362,8 +362,9 @@ void Server::dispatch(const cref_t<Message> &m)
     handle_peer_request(ref_cast<MMDSPeerRequest>(m));
     return;
   default:
-    derr << "server unknown message " << m->get_type() << dendl;
-    ceph_abort_msg("server unknown message");  
+    derr << "server unknown message: " << m->get_type() << ", therefore closing session and blocklisting client " << session->get_client() << dendl;
+    CachedStackStringStream css;
+    mds->evict_client(session->get_client().v, false, true, *css, nullptr);
```

Comment:
For the same reason you can't do the cast above, this is not correct. You can't evict a peer MDS.

Reply:
Isn't the client that will be evicted here? I'm using MDSRank::evict_client for this purpose

Reply:
No. This is the default for the switch statement. So if an MDS sends an unknown message, you'll try to evict it as a client.

Reply:
Got it, I've made some changes PTAL.